pkg/odo/cli/url/describe.go

```diff
@@ -61,7 +67,42 @@ func (o *URLDescribeOptions) Validate() (err error) {
 
 // Run contains the logic for the odo url list command
 func (o *URLDescribeOptions) Run() (err error) {
-	if experimental.IsExperimentalModeEnabled() {
+	if pushtarget.IsPushTargetDocker() && experimental.IsExperimentalModeEnabled() {
+		client, err := lclient.New()
+		if err != nil {
+			return err
+		}
+		u, err := url.GetContainerURL(client, o.EnvSpecificInfo, o.url, o.EnvSpecificInfo.GetName())
+		if err != nil {
+			return err
+		}
+
+		if log.IsJSON() {
+			machineoutput.OutputSuccess(u)
+		} else {
+			tabWriterURL := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+			fmt.Fprintln(tabWriterURL, "NAME", "\t", "STATE", "\t", "URL", "\t", "PORT")
+			var urlString string
+			if u.Status.State == url.StateTypeNotPushed {
+				// to be consistent with URL for ingress and routes
+				// if not pushed, display URl as ://
+				urlString = "://"
+			} else {
+				urlString = u.Spec.Host + ":" + strconv.Itoa(u.Spec.ExternalPort)
+			}
+			// are there changes between local and cluster states?
+			outOfSync := false
+			fmt.Fprintln(tabWriterURL, u.Name, "\t", u.Status.State, "\t", urlString, "\t", u.Spec.Port)
+			if u.Status.State != url.StateTypePushed {
+				outOfSync = true
+			}
+			tabWriterURL.Flush()
+			if outOfSync {
+				fmt.Fprintf(os.Stdout, "\n")
+				fmt.Fprintf(os.Stdout, "There are local changes. Please run 'odo push'.\n")
```

Comment:
using straight up `os.Stdout` is a big no-no, it's a big reason why we implemented: https://github.com/openshift/odo/blob/master/pkg/log/status.go#L338

Please use `log.Infof` for the output.

Also, if it's out of sync, wouldn't it make sense to error out and ask to run `odo push` ?

Reply:
This is how odo1 behaves. See: https://github.com/openshift/odo/blob/9411894b1b50ea0caf7f09e7810848e6997030ef/pkg/odo/cli/url/describe.go#L99-L102

If it's out of sync, means the url is "not pushed" or "locally deleted". We need to tell user they need to do a `Push` to get the changes reflected, but It's definitely not an error,.